test/gtest/ucp/test_ucp_peer_failure.cc

```diff
@@ -474,3 +474,90 @@ UCS_TEST_SKIP_COND_P(test_ucp_peer_failure, disable_sync_send,
         EXPECT_EQ(UCS_ERR_UNSUPPORTED, UCS_PTR_STATUS(req));
     }
 }
+
+class test_ucp_peer_failure_keepalive : public test_ucp_peer_failure
+{
+public:
+    void init() {
+        test_ucp_peer_failure::init();
+        sender().connect(&receiver(), get_ep_params());
+        receiver().connect(&sender(), get_ep_params());
+    }
+
+    void kill_receiver()
+    {
+        ucp_ep_h ep = receiver().revoke_ep(0, 0);
+
+        void *creq = ucp_ep_close_nb(ep, UCP_EP_CLOSE_MODE_FORCE);
+        request_wait(creq);
+    }
+
+    void sendrecv()
+    {
+        char send_buf[256];
+        char recv_buf[256];
+
+        ucp_request_param_t send_param; 
+        ucp_request_param_t recv_param; 
+
+        send_param.op_attr_mask = 0;
+        recv_param.op_attr_mask = 0;
+
+        void *recv_req = ucp_tag_recv_nbx(receiver().worker(), recv_buf,
+                                          sizeof(recv_buf),
+                                          0, 0, &recv_param);
+        void *send_req = ucp_tag_send_nbx(sender().ep(), send_buf,
+                                          sizeof(send_buf),
+                                          0, &send_param);
+        request_wait(recv_req);
+        request_wait(send_req);
+    }
+
+    static std::vector<ucp_test_param>
+    enum_test_params(const ucp_params_t& ctx_params, const std::string& name,
+                     const std::string& test_case_name, const std::string& tls)
+    {
+        std::vector<ucp_test_param> result;
+
+        ucp_params_t params = ucp_test::get_ctx_params();
+
+        params.field_mask  |= UCP_PARAM_FIELD_FEATURES;
+
+        params.features = UCP_FEATURE_TAG;
+        generate_test_params_variant(params, name, test_case_name + "/tag", tls,
+                                     UCP_FEATURE_TAG, result);
+
+        return result;
+    }
+};
+
+UCS_TEST_P(test_ucp_peer_failure_keepalive, kill_receiver,
+           "KEEPALIVE_TIMEOUT=2us", "KEEPALIVE_NUM_EPS=inf") {
+    if (ucp_ep_config(sender().ep())->key.ep_check_map == 0) {
```

Comment:
should it be in the `init()`, but since it will throw a SKIP exception, we should take care of objects created during `init()` (i.e. release them, since cleanup() won't be called).
we will add a new test and we should check this there too

Reply:
for this particular test - yes, but in general case there could be tests where detected missing keepalive calls when no lanes support it

Reply:
in this case it should be skipped by UNREACHABLE error from wireup

Reply:
no, another test, for example, is test for keepalive is not called for EP which doesn't enabled keepalive